### ISSUE TITLE
RARBG still has errors

### ISSUE THREAD

**Issue by ukharley** [Jackett/Jackett#4104]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue. Duplicating issues makes it more difficult for everyone to follow. Your issue may have already been solved in the past as well.

Provide a description of the feature request or bug, the more details the better.

If you are experiencing an issue with a tracker, then:
1. Use your browser to check you can access the site directly, and if a login is required, check you can login and that you do not have any outstanding account issues.
2. If you haven't already, try upgrading to the latest version of Jackett, your issue may have already been resolved.
3. If it is still not working for you, then a **full enhanced log must be included**. Instructions for obtaining logs are here: https://github.com/Jackett/Jackett#troubleshooting

**Jackett version**:  Version 0.10.398.0
**Mono version** (if not using Windows):  Using Windows.

-------------------------------
Still get this in Medusa: Torrent file content is empty

Couldn't attach log file, so pasting relevant part of Jackett enhanced log:

```
2018-10-31 10:31:10.4171 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/dl/rarbg/?jackett_apikey=*****&path=Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5&file=The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]   
2018-10-31 10:31:10.4341 Info Route matched with {action = "Download", controller = "Download"}. Executing action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) 
2018-10-31 10:31:10.4341 Info Executing action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole) with arguments (rarbg, Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5, ************, The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]) - Validation state: Valid 
2018-10-31 10:31:12.6722 Error Error downloading rarbg https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 System.Exception: Downlaod Failed, expected redirect
   at Jackett.Common.Indexers.Rarbg.<Download>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Jackett.Server.Controllers.DownloadController.<Download>d__5.MoveNext()
2018-10-31 10:31:12.6722 Info Executed action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole), returned result Microsoft.AspNetCore.Mvc.NotFoundResult in 2233.2265ms. 
2018-10-31 10:31:12.6882 Info Executing HttpStatusCodeResult, setting HTTP status code 404 
2018-10-31 10:31:12.6882 Info Executed action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) in 2254.146ms 
2018-10-31 10:31:12.7012 Info Request finished in 2276.2193ms 404  

```

**Comment by kaso17** [comment]
Please enable enhanced logging and post an updated log showing a failed download.
Please also try to redownload that particular torrent.

**Comment by ukharley** [comment]
Still the same.

Trying to attach log gives error: Something went really wrong.......

Relevant portion on **advanced logging** below:
```
2018-10-31 15:46:18.9607 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/dl/rarbg/?jackett_apikey=*************************&path=Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5&file=The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]   
2018-10-31 15:46:18.9687 Info Route matched with {action = "Download", controller = "Download"}. Executing action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) 
2018-10-31 15:46:18.9687 Info Executing action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole) with arguments (rarbg, Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5, *************************, The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]) - Validation state: Valid 
2018-10-31 15:46:19.2519 Error Error downloading rarbg https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 System.Exception: Downlaod Failed, expected redirect
   at Jackett.Common.Indexers.Rarbg.<Download>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Jackett.Server.Controllers.DownloadController.<Download>d__5.MoveNext()
2018-10-31 15:46:19.2599 Info Executed action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole), returned result Microsoft.AspNetCore.Mvc.NotFoundResult in 275.9252ms. 
2018-10-31 15:46:19.2599 Info Executing HttpStatusCodeResult, setting HTTP status code 404 
2018-10-31 15:46:19.2759 Info Executed action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) in 299.2019ms 
2018-10-31 15:46:19.2759 Info Request finished in 322.1325ms 404  
2018-10-31 15:46:19.2759 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/dl/rarbg/?jackett_apikey=*************************&path=Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5&file=The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]   
2018-10-31 15:46:19.2969 Info Route matched with {action = "Download", controller = "Download"}. Executing action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) 
2018-10-31 15:46:19.2969 Info Executing action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole) with arguments (rarbg, Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5, *************************, The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]) - Validation state: Valid 
2018-10-31 15:46:21.6182 Error Error downloading rarbg https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 System.Exception: Downlaod Failed, expected redirect
   at Jackett.Common.Indexers.Rarbg.<Download>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Jackett.Server.Controllers.DownloadController.<Download>d__5.MoveNext()
2018-10-31 15:46:21.6262 Info Executed action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole), returned result Microsoft.AspNetCore.Mvc.NotFoundResult in 2314.1931ms. 
2018-10-31 15:46:21.6262 Info Executing HttpStatusCodeResult, setting HTTP status code 404 
2018-10-31 15:46:21.6402 Info Executed action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) in 2335.7561ms 
2018-10-31 15:46:21.6402 Info Request finished in 2358.0359ms 404  
```

**Comment by kaso17** [comment]
It's still a normal log (did you click on `apply server settings` after enabling it?).
The download links provided via the torznab feed to medusa and via the jackett manual search are the same. So it's very odd.

**Comment by ukharley** [comment]
Updated log:
```
2018-10-31 16:26:59.4544 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/dl/rarbg/?jackett_apikey=fdq0meafvekb8eyueapxm5mgokgdvnhl&path=Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5&file=The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]   
2018-10-31 16:26:59.4544 Info Route matched with {action = "Download", controller = "Download"}. Executing action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) 
2018-10-31 16:26:59.4644 Info Executing action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole) with arguments (rarbg, Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5, fdq0meafvekb8eyueapxm5mgokgdvnhl, The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]) - Validation state: Valid 
2018-10-31 16:26:59.4644 Debug WebClient(HttpWebClient).GetString(Url:https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63) 
2018-10-31 16:26:59.7657 Debug WebClient(HttpWebClient): Returning OK => Invalid token. Use get_token for a new one! 
2018-10-31 16:26:59.7657 Error Error downloading rarbg https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 System.Exception: Downlaod Failed, expected redirect
   at Jackett.Common.Indexers.Rarbg.<Download>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Jackett.Server.Controllers.DownloadController.<Download>d__5.MoveNext()
2018-10-31 16:26:59.7776 Info Executed action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole), returned result Microsoft.AspNetCore.Mvc.NotFoundResult in 309.12ms. 
2018-10-31 16:26:59.7776 Info Executing HttpStatusCodeResult, setting HTTP status code 404 
2018-10-31 16:26:59.7916 Info Executed action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) in 326.9752ms 
2018-10-31 16:26:59.7916 Info Request finished in 343.2392ms 404  
2018-10-31 16:26:59.8066 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/dl/rarbg/?jackett_apikey=fdq0meafvekb8eyueapxm5mgokgdvnhl&path=Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5&file=The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]   
2018-10-31 16:26:59.8066 Info Route matched with {action = "Download", controller = "Download"}. Executing action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) 
2018-10-31 16:26:59.8066 Info Executing action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole) with arguments (rarbg, Q2ZESjhMNVlSeVQ1T0QxQnVheHlpSU93QmtpYVJvTXUxZjJ0XzE0c1NoU24wd3dMMFR4czRSQkpfb2xiN0M2aVVhaE02SWFXdHRhdVpXOEs4R21qSThfU2NlSzEzY2Z4ZUpmeVJBRm1pWnhSVjdvUXBpa2lrZ1RDc0lUOWpMaUJPRjFlRGtiT0xSeEZoTHB0czNJd0E1UnRYRjlyeEw2aWl0Q3ZqWXozTDVUSk9lZGxrUmNwSnhHOExyM1dSZm5ZSVpPYlFoYURCV3RpdUxrQWZQVUpoWHhBN2E0ajFRSi1NWlA0dDlsOFBlUzlPTVl5, fdq0meafvekb8eyueapxm5mgokgdvnhl, The.Kids.Are.Alright.S01E03.720p.HDTV.x264-KILLERS[rartv]) - Validation state: Valid 
2018-10-31 16:26:59.8266 Debug WebClient(HttpWebClient).GetString(Url:https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63) 
2018-10-31 16:26:59.8266 Debug WebClient(HttpWebClient): delaying request for https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 by 2.0330128 seconds 
2018-10-31 16:27:02.1336 Debug WebClient(HttpWebClient): Returning OK => Invalid token. Use get_token for a new one! 
2018-10-31 16:27:02.1396 Error Error downloading rarbg https://torrentapi.org/redirect_to_info.php?token=e3kbwfizoa&p=1_6_7_6_3_6_0__d3c975fa63 System.Exception: Downlaod Failed, expected redirect
   at Jackett.Common.Indexers.Rarbg.<Download>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Jackett.Server.Controllers.DownloadController.<Download>d__5.MoveNext()
2018-10-31 16:27:02.1396 Info Executed action method Jackett.Server.Controllers.DownloadController.Download (JackettConsole), returned result Microsoft.AspNetCore.Mvc.NotFoundResult in 2319.7545ms. 
2018-10-31 16:27:02.1516 Info Executing HttpStatusCodeResult, setting HTTP status code 404 
2018-10-31 16:27:02.1516 Info Executed action Jackett.Server.Controllers.DownloadController.Download (JackettConsole) in 2339.3406ms 
2018-10-31 16:27:02.1516 Info Request finished in 2358.7118ms 404  
2018-10-31 16:27:20.1568 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/api/v2.0/server/logs?_=1541002994400   
2018-10-31 16:27:20.1568 Info Route matched with {action = "Logs", controller = "ServerConfiguration"}. Executing action Jackett.Server.Controllers.ServerConfigurationController.Logs (JackettConsole) 
2018-10-31 16:27:20.1698 Info Authorization was successful. 
2018-10-31 16:27:20.1698 Info Executing action method Jackett.Server.Controllers.ServerConfigurationController.Logs (JackettConsole) - Validation state: Valid 
2018-10-31 16:27:20.1858 Info Executed action method Jackett.Server.Controllers.ServerConfigurationController.Logs (JackettConsole), returned result Microsoft.AspNetCore.Mvc.ObjectResult in 0.0046ms. 
2018-10-31 16:27:20.1858 Info Executing ObjectResult, writing value of type 'System.Collections.Generic.List`1[[Jackett.Common.Models.CachedLog, Jackett.Common, Version=0.10.398.0, Culture=neutral, PublicKeyToken=null]]'. 
2018-10-31 16:27:20.2038 Info Executed action Jackett.Server.Controllers.ServerConfigurationController.Logs (JackettConsole) in 34.2085ms 
2018-10-31 16:27:20.2038 Info Request finished in 54.1067ms 200 application/json; charset=utf-8 
```

**Comment by kaso17** [comment]
ok, thank you. Looks like there's to much delay between the search and the actual torrent file download resulting in an expired token. I'll add some token refresh logic to the download too.

**Comment by kaso17** [comment]
Should be fixed via fd0c92f4397ad2881a3ed5a97e2278a0187596c1.
Please test version v0.10.410
